File: Sif/src/app/core/models/articleCategory.ts
import { Article,  } from './article';
import { Category } from './category';

export class ArticleCategory {
  constructor(article: Article, category: Category) {
    this.articleId = article.articleId;
    this.categoryId = category.categoryId;
  }

  articleId?: number;
  categoryId?: number;
}

File: Thor.Models/Database/ArticleCategory.cs
using DTO = Thor.Models.Dto;

namespace Thor.Models.Database;

public class ArticleCategory
{
  public int ArticleId { get; set; }
  public int CategoryId { get; set; }

  public Article Article { get; set; }
  public Category Category { get; set; }

  public ArticleCategory() {}
  public ArticleCategory(DTO.ArticleCategory articleCategory) {
    ArticleId = articleCategory.ArticleId;
    CategoryId = articleCategory.CategoryId;
  }
}